lib/expression.ts

```diff
@@ -20,22 +20,83 @@ export default class ExpressionTranspiler extends base.TranspilerBase {
           this.visit(binExpr.right);
```

Comment:
Merge all of this into one switch statement, `if` on some expression followed by `switch` on the same expression is odd.